src/lib.rs

```diff
@@ -79,12 +84,45 @@ pub trait IsExecutable {
     fn is_executable(&self) -> bool;
 }
 
+/// Returns `Result<Path, io::Error>` if there is a file at the given path and the
+/// current run-level is permitted to execute it.
+///
+/// See the module documentation for details.
+#[cfg(unix)]
+pub fn is_permitted<P>(path: P) -> Result<::std::path::PathBuf, io::Error>
```

Comment:
Use the full path for `io` here and get rid of the import on line 58, it's unneeded unless compiling for Unix